app/src/main/java/ch/epfl/sdp/game/PreyActivity.kt

```diff
@@ -1,132 +1,134 @@
+                        break
+                    }
                 }
             }
-        })
-        binding.repeatLastLocation.setOnClickListener(View.OnClickListener {
-            if (ActivityCompat.checkSelfPermission(applicationContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(applicationContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                return@OnClickListener
-            }
-            val last = mLocationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER)
-            if (last != null) {
-                locationSynchronizer.updateOwnLocation(ch.epfl.sdp.game.data.Location(last.latitude, last.longitude))
-            } else {
-                Toast.makeText(applicationContext, "No last known location", Toast.LENGTH_SHORT).show()
-            }
-        })
+        }
+
+        predatorRadarFragment.updateInfos(mostDangerousManDistance, rangePopulation)
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String?>, grantResults: IntArray) {
-        if (requestCode == MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION) {
-            enableRequestUpdates()
+    private fun resetRange() {
+        for(i in ranges.indices) {
+            rangePopulation[ranges[i]] = 0
         }
     }
 
-    private fun enableRequestUpdates() {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            if (checkSelfPermission(Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && checkSelfPermission(Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
-                        MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION)
-                return
-            }
+    override fun onLocationChanged(newLocation: Location) {
+        updateThreat()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        if (validGame) {
+            locationHandler.stop()
         }
-        locationSynchronizer = SimpleLocationSynchronizer(0, mPlayerID, pubSub)
-        mLocationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, LOCATION_REFRESH_TIME.toLong(), LOCATION_REFRESH_DISTANCE.toFloat(), mLocationListener)
     }
 
-    private fun disableRequestUpdates() {
-        mLocationManager.removeUpdates(mLocationListener)
+    override fun onStatusChanged(provider: String, status: Int, extras: Bundle) {
     }
 
-    companion object {
-        private const val LOCATION_REFRESH_DISTANCE = 5
-        private const val LOCATION_REFRESH_TIME = 1000
-        private const val MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION = 10
+    override fun onProviderEnabled(provider: String) {
     }
+
+    override fun onProviderDisabled(provider: String) {
```

Comment:
Are empty functions necessary ?

Reply:
They may be needed in some other implementation so i'd let them